resonance/db.py

```diff
@@ -86,7 +106,71 @@ def combine(*blocks):
     @staticmethod
     def make_empty(si):
         obj = np.empty((0, si.channels)).view(Channels)
-        ts =  np.array([], dtype=np.int64)
+        ts = np.array([], dtype=np.int64)
+        Base.__new__(obj, si, ts)
+        return obj
+
+
+class SingleWindow(Base, np.ndarray):
+    def __new__(cls, ts, data):
+        obj = np.asarray(data).view(SingleWindow)
+
+        if isinstance(ts, float) or isinstance(ts, int):
+            ts = np.asarray([ts], dtype=np.longlong)
+
+        Base.__new__(obj, ts, data)
+        return obj
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __eq__(self, other):
+        if isinstance(other, SingleWindow):
+            return np.array_equal(self._ts, other._ts) and np.array_equal(self, other)
+        else:
+            return np.ndarray.__eq__(self, other)
+
+
+class Window(Base, np.ndarray):
+    def __new__(cls, si, ts, data):
+
+        if isinstance(ts, int) or isinstance(ts, float):
+            ts = ts - np.flip(np.arange(0, np.size(data, 0) / si.channels)) * 1E9/si.samplingRate
+
+        obj = np.array(SingleWindow(ts, data)).view(Window)
+        if len(obj.shape) != 2 or np.size(obj, 1) != si.channels:
```

Comment:
I say here you check the size and if it is not required size you throw, and move that check to the first line of __new__